### PR TITLE
intel-oneapi-mpi virtual provider support

### LINKED ISSUE #20725: Installation issue: intel-oneapi-mpi is incomplete
### Steps to reproduce the issue

```console
$ (spack) rscohn2@rscohn1-MOBL:work$ spack install hdf5%intel@2021.1 ^intel-oneapi-mpi
[+] /home/rscohn2/local/Projects/spack/work/opt/spack/linux-ubuntu20.04-skylake/intel-2021.1/intel-oneapi-mpi-2021.1.1-4ipltnk6vi7vm\
oezshrfaysyuaii6ip2
[+] /home/rscohn2/local/Projects/spack/work/opt/spack/linux-ubuntu20.04-skylake/intel-2021.1/zlib-1.2.11-zkxwt5sqzpfpsitx65hqveu3lzw\
rmg5u
==> Installing hdf5-1.10.7-ytdsc3yfdrwemyaaajntq4vnnoyzvdx2
==> No binary for hdf5-1.10.7-ytdsc3yfdrwemyaaajntq4vnnoyzvdx2 found: installing from source
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/7a/7a1a0a54371275ce2dfc5cd093775bb025c365846\
512961e7e5ceaecb437ef15.tar.gz
############################################################################################################################# 100.0%
==> hdf5: Executing phase: 'autoreconf'
==> hdf5: Executing phase: 'configure'
==> Error: AttributeError: 'Spec' object has no attribute 'mpicc'

/home/rscohn2/local/Projects/spack/work/var/spack/repos/builtin/packages/hdf5/package.py:303, in configure_args:
        300            # interface. Since they are still accessible via the C interface,
        301            # this is not actually a problem.
        302            extra_args += ['--enable-parallel',
  >>    303                           'CC=%s' % self.spec['mpi'].mpicc]
        304
        305            if '+cxx' in self.spec:
        306                extra_args.append('CXX=%s' % self.spec['mpi'].mpicxx)

See build log for details:
  /tmp/rscohn2/spack-stage/spack-stage-hdf5-1.10.7-ytdsc3yfdrwemyaaajntq4vnnoyzvdx2/spack-build-out.txt

(spack) rscohn2@rscohn1-MOBL:work$
...
```

### PR DESCRIPTION
**Pull request by rscohn2**
resolves #20725 
@scheibelp @frankwillmore: FYI
I installed hdf5%oneapi ^intel-oneapi-mpi and
hdf5 ^intel-oneapi-mpi
and verified that it uses the right compiler in each case.

### CODE REVIEW

**Review by wspear — APPROVED** [review]
This fixed the issue for me with no observable side effects.

**Comment by rscohn2** [comment]
@wspear : Thanks for checking.

File: var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py [hunk 1]
```diff
@@ -28,6 +28,20 @@ def __init__(self, spec):
...
 
+    def setup_dependent_package(self, module, dep_spec):
+        dir = join_path(self.prefix, 'mpi', 'latest', 'bin')
+        self.spec.mpicc  = join_path(dir, 'mpicc')
+        self.spec.mpicxx = join_path(dir, 'mpicxx')
+        self.spec.mpif77 = join_path(dir, 'mpif77')
+        self.spec.mpifc  = join_path(dir, 'mpifc')
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.set('MPICH_CC', spack_cc)
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py:39]: Do the `intel-oneapi-mpi` MPI compiler wrappers use these `MPICH`-prefixed variables? I would have guessed they would use something like `I_MPI_CC` (which is set by `spack.build_systems.intel.mpi_setup_dependent_build_environment`) etc.

Note that if the `mpicc` wrappers provided by this package don't use such a variable, then you don't have to set it at all. But as mentioned in https://github.com/spack/spack/issues/3487#issuecomment-287564230, these wrappers might use such a variable to locate e.g. the underlying CC compiler that should be used.

Comment (rscohn2) [reply var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py:39]: They use the MPICH prefixed variables. That may not have been true for pre-oneapi releases of Intel MPI.

```
(spack) rscohn2@rscohn1-MOBL:intel-oneapi-mpi$ which mpicc
/opt/intel/oneapi/mpi/2021.1.1//bin/mpicc
(spack) rscohn2@rscohn1-MOBL:intel-oneapi-mpi$ mpicc -show test.c
gcc 'test.c' -I'/opt/intel/oneapi/mpi/2021.1.1/include' -L'/opt/intel/oneapi/mpi/2021.1.1/lib/release' -L'/opt/intel/oneapi/mpi/2021\
.1.1/lib' -Xlinker --enable-new-dtags -Xlinker -rpath -Xlinker '/opt/intel/oneapi/mpi/2021.1.1/lib/release' -Xlinker -rpath -Xlinker\
 '/opt/intel/oneapi/mpi/2021.1.1/lib' -lmpifort -lmpi -lrt -lpthread -Wl,-z,now -Wl,-z,relro -Wl,-z,noexecstack -Xlinker --enable-ne\
w-dtags -ldl
(spack) rscohn2@rscohn1-MOBL:intel-oneapi-mpi$ MPICH_CC=icx mpicc -show test.c
icx 'test.c' -I'/opt/intel/oneapi/mpi/2021.1.1/include' -L'/opt/intel/oneapi/mpi/2021.1.1/lib/release' -L'/opt/intel/oneapi/mpi/2021\
.1.1/lib' -Xlinker --enable-new-dtags -Xlinker -rpath -Xlinker '/opt/intel/oneapi/mpi/2021.1.1/lib/release' -Xlinker -rpath -Xlinker\
 '/opt/intel/oneapi/mpi/2021.1.1/lib' -lmpifort -lmpi -ldl -lrt -lpthread
(spack) rscohn2@rscohn1-MOBL:intel-oneapi-mpi$
```

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**